components/samsung_ac/protocol_nasa.cpp

```diff
@@ -749,7 +870,7 @@ namespace esphome
 
             target->register_address(source);
 
-            if (debug_log_packets)
+            if (debug_log_messages)

```

Comment:
This did not do what you would like to archive. This also print the undefined messages.